packages/expo-sqlite/src/next/Database.ts

```diff
@@ -0,0 +1,177 @@
+import { EventEmitter, Subscription } from 'expo-modules-core';
+
+import ExpoSQLite from './ExpoSQLiteNext';
+import { NativeDatabase, OpenOptions } from './NativeDatabase';
+import { BindParams, RunResult, Statement, VariadicBindParams } from './Statement';
+
+export { OpenOptions };
+
+const emitter = new EventEmitter(ExpoSQLite);
+
+/**
+ * A SQLite database.
+ */
+export class Database {
+  constructor(private readonly nativeDatabase: NativeDatabase) {}
+
+  /**
+   * Synchronous call to return whether the database is currently in a transaction.
+   */
+  public isInTransaction(): boolean {
+    return this.nativeDatabase.isInTransaction();
+  }
+
+  /**
+   * Asynchronous call to return whether the database is currently in a transaction.
+   */
+  public isInTransactionAsync(): Promise<boolean> {
+    return this.nativeDatabase.isInTransactionAsync();
+  }
+
+  /**
+   * Close the database.
+   */
+  public closeAsync(): Promise<void> {
+    return this.nativeDatabase.closeAsync();
+  }
+
+  /**
+   * Execute all SQL queries in the supplied string.
+   * > Note: The queries are not escaped for you! Be careful when constructing your queries.
+   *
+   * @param source A string containing all the SQL queries.
+   */
+  public execAsync(source: string): Promise<void> {
+    return this.nativeDatabase.execAsync(source);
+  }
+
+  /**
+   * Prepare a SQL statement.
+   *
+   * @param source A string containing the SQL query.
+   * @returns A `Statement` object.
+   */
+  public async prepareAsync(source: string): Promise<Statement> {
+    const nativeStatement = new ExpoSQLite.NativeStatement();
+    await this.nativeDatabase.prepareAsync(nativeStatement, source);
+    return new Statement(this.nativeDatabase, nativeStatement);
+  }
+
+  /**
+   * Execute a transaction and automatically commit/rollback based on the `txn` success.
+   *
+   * @param txn An async function to execute within a transaction.
+   */
+  public async transactionAsync(txn: () => Promise<void>): Promise<void> {
+    try {
+      await this.nativeDatabase.execAsync('BEGIN');
+      await txn();
+      await this.nativeDatabase.execAsync('COMMIT');
+    } catch (e) {
+      await this.nativeDatabase.execAsync('ROLLBACK');
+      throw e;
+    }
+  }
+
+  //#region Statement API shorthands
+
+  /**
+   * Shorthand for `prepareAsync` and `Statement.runAsync`.
+   * Unlike `Statement.runAsync`, this method finalizes the statement after execution.
+   *
+   * @param source A string containing the SQL query.
+   * @param params Parameters to bind to the query.
+   */
+  public runAsync(source: string, ...params: VariadicBindParams): Promise<RunResult>;
```

Comment:
I'm wondering if this will render correctly in our docs 🤔

Reply:
looks fine. please see #24962 and preview at http://docs.expo.dev-pr-24962.s3-website-us-east-1.amazonaws.com/versions/unversioned/sdk/sqlite-next/